vulnerabilities/importers/mattermost.py

```diff
@@ -0,0 +1,194 @@
+MM_REPO = {
+    "Mattermost Mobile Apps": "mattermost/mattermost-mobile",
+    "Mattermost Server": "mattermost/mattermost-server",
+    "Mattermost Desktop App": "mattermost/desktop",
+}
+
+
+class MattermostDataSource(DataSource):
+    def updated_advisories(self):
+        # FIXME: Change after this https://forum.mattermost.org/t/mattermost-website-returning-403-when-headers-contain-the-word-python/11412
+        self.set_api()
+        data = requests.get(
+            SECURITY_UPDATES_URL, headers={"user-agent": "aboutcode/vulnerablecode"}
+        ).content
+        return self.batch_advisories(self.to_advisories(data))
+
+    def set_api(self):
+        self.version_api = GitHubTagsAPI()
+        asyncio.run(
+            self.version_api.load_api(
+                [
+                    MM_REPO["Mattermost Mobile Apps"],
+                    MM_REPO["Mattermost Server"],
+                    MM_REPO["Mattermost Desktop App"],
+                ]
+            )
+        )
+
+    def to_advisories(self, data):
+        advisories = []
+        soup = BeautifulSoup(data, features="lxml")
+        for row in soup.table.tbody.find_all("tr"):
+            (
+                ref_col,
+                severity_col,
+                affected_col,
+                _,
+                fixed_col,
+                desc_col,
+                name_col,
+            ) = row.select("td")
+
+            name = name_col.text.strip()
+            if name not in MM_REPO:
+                continue
+
+            fixed_versions = split_versions(fixed_col.text)
+            fixed_packages = [
+                PackageURL(
+                    type="mattermost",
+                    name=name,
+                    version=version,
+                )
+                for version in fixed_versions
+            ]
+
+            (
+                affected_version_ranges,
+                excluded_version_ranges,
+            ) = to_affected_version_ranges(affected_col.text, fixed_col.text)
+
+            affected_packages = [
+                PackageURL(type="mattermost", name=name, version=version)
+                for version in self.version_api.get(MM_REPO[name])
+                if
+                # The versions comparisions and advisories are not compatible with cloud-* versions
+                not version.startswith("cloud-")
+                and any((version in version_range for version_range in affected_version_ranges))
+                and not any((version in version_range for version_range in excluded_version_ranges))
+            ]
+
+            """
+            Severities are either "na" or cvssv3.1_qr
+            """
+            references = [
+                Reference(
+                    reference_id=ref_col.text,
+                    url=SECURITY_UPDATES_URL,
+                    severities=[
+                        VulnerabilitySeverity(
+                            system=scoring_systems["cvssv3.1_qr"], value=severity_col.text
```

Comment:
I don't think it's a `cvssv3.1_qr`.  IMHO we should create a scoring system just for these.

Reply:
The severities are Low, Medium and High which matches the list given @ https://www.first.org/cvss/specification-document#Qualitative-Severity-Rating-Scale

Reply:
Implemented in #415